core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/stacks/cluster/CloudStorageConverter.java

```diff
@@ -433,7 +433,7 @@ private S3Identity identityRequestToS3(StorageIdentityBase storageIdentityReques
 
     private EfsIdentity identityRequestToEfs(StorageIdentityBase storageIdentityRequest) {
         EfsIdentity efsIdentity = new EfsIdentity();
-        efsIdentity.setInstanceProfile(storageIdentityRequest.getS3().getInstanceProfile());
+        efsIdentity.setInstanceProfile(storageIdentityRequest.getEfs().getInstanceProfile());
```

Comment:
Out of curiosity, what is the purpose of this EC2 instance profile? Where is it used?

Reply:
This will be used for access control of the EFS.

Reply:
I surely know what an instance profile is generally meant to provide. 😃 So, let me rephrase: FreeIPA, DL and DH EC2 instances always have either of two EC2 instance profiles attached (exactly one, as it is not possible to attach multiple ones): the Log profile/role or the IDBroker Assumer profile/role. So I am wondering where the access rights for EFS come into play. Are you planning to extend these roles with EFS permissions (presumably the Log one, as I guess EFS will not be attached to IDBroker nodes)?

Reply:
So access to EFS file system instances seems to be [governed by file system policies](https://docs.aws.amazon.com/efs/latest/ug/iam-access-control-nfs-efs.html). I am not sure if any additional permissions are needed for the EC2 instance itself. Have you possibly verified this?

Reply:
this is definitely an additional thing to support this solution. Even the current roles are hard to understand and it seems now we need one more.

Reply:
I am not planning to attach EFS to IDBroker nodes. 
Since EC2 instance can only attach one instance profile, I will need to extend those roles with EFS permissions.

Reply:
I have discussed with Lajos in slack. I will handle this in another jira. So close this discussion for now.

Reply:
As discussed offline, it is unclear what kind of permissions are needed for EC2 instances in order to utilize EFS mounts. IDBroker and RAZ are out of the game, as the mounting logic must not depend on either Hadoop or any specific user. More elaboration and experimentation is needed to clarify these, but that may be covered in a subsequent PR.